frappe/utils/nestedset.py

```diff
@@ -149,7 +150,16 @@ def rebuild_tree(doctype, parent_field):
 	frappe.db.auto_commit_on_many_writes = 1
 
 	right = 1
-	result = frappe.db.sql("SELECT name FROM `tab%s` WHERE `%s`='' or `%s` IS NULL ORDER BY name ASC" % (doctype, parent_field, parent_field))
+	doctype = DocType(doctype)
+	result = (
+		frappe.qb.from_(doctype)
+		.where(
+			(getattr(doctype, parent_field) == "") | (parent_field.isnull())
```

Comment:
<img width="1440" alt="Screenshot 2021-11-03 at 3 56 01 PM" src="https://user-images.githubusercontent.com/36654812/140044274-b4765444-54b4-4643-aeec-3504955859d8.png">

This doesn't seem to work. Anything else also doesn't seem to work. Says `TypeError: 'Field' object is not callable` for the most part....PyPika doesn't have provision for dynamically setting Columns? Tried `Column`, `getattr`, `.get` and `Term`.

---

Probably not a good idea to re-define doctype...